rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -254,27 +255,23 @@ public class RskContext implements NodeContext, NodeBootstrapper {
     private volatile boolean closed;
 
     /***** Constructors ***********************************************************************************************/
-
     public RskContext(String[] args) {
-        this(args, false);
-    }
-
-    public RskContext(String[] args, boolean ignoreUnmatchedArgs) {
-        this(new CliArgs.Parser<>(
-                NodeCliOptions.class,
-                NodeCliFlags.class,
-                ignoreUnmatchedArgs
-        ).parse(args));
+        RskCli rskCli = new RskCli();
+        rskCli.load(args);
+        initialize(rskCli, rskCli.getCliArgs());
     }
 
-    private RskContext(CliArgs<NodeCliOptions, NodeCliFlags> cliArgs) {
+    private void initialize(RskCli rskCli, CliArgs<NodeCliOptions, NodeCliFlags> cliArgs) {
+        Objects.requireNonNull(rskCli, "RskjCli must not be null");
         this.cliArgs = cliArgs;
         initializeNativeLibs();
     }
 
     /***** Public Methods *********************************************************************************************/
 
-    public CliArgs<NodeCliOptions, NodeCliFlags> getCliArgs() {
+    public synchronized CliArgs<NodeCliOptions, NodeCliFlags> getCliArgs() {
```

Comment:
`cliArgs` is set only during construction. Seems no need to mark this method as `synchronized` and call `checkIfNotClosed()`